### PR TITLE
pyth oracle

### PR DESCRIPTION
**Pull request by JChiaramonte7**

### CODE REVIEW

File: src/Account.sol [hunk 1]
```diff
@@ -930,13 +946,26 @@ contract Account is IAccount, Auth, OpsReady {
...
+            PERPS_V2_EXCHANGE_RATE.resolveAndGetLatestPrice(assetId);
+
+        // resolveAndGetLatestPrice is provide by pyth
+        PriceOracleUsed priceOracle = PriceOracleUsed.PYTH;
+
+        // if the price is stale, get the latest price from the market
+        // (i.e. Chainlink provided price)
+        if (publishTime < block.timestamp - MAX_PRICE_LATENCY) {
+            priceOracle = PriceOracleUsed.CHAINLINK;
+            (price,) = _market.assetPrice();
```
Comment (tommyrharper) [inline src/Account.sol:964]: **Issue:** I think you need to handle the case here where `assetPrice()` returns an invalid price:
```solidity
        (uint256 price, bool invalid) = _market.assetPrice();
        if (invalid) {
            revert InvalidPrice();
        }
```

**Review by tommyrharper — CHANGES_REQUESTED** [review]
Looks all good to me except checking if the `_market.assetPrice()` returns `invalid` and handling that.

File: src/Account.sol [hunk 2]
```diff
@@ -755,6 +769,10 @@ contract Account is IAccount, Auth, OpsReady {
         /// @dev will revert if task id does not exist {Automate.cancelTask: Task not found}
         IOps(OPS).cancelTask({taskId: conditionalOrder.gelatoTaskId});
 
+        // pay Gelato imposed fee for conditional order execution
+        (uint256 fee, address feeToken) = IOps(OPS).getFeeDetails();
+        _transfer({_amount: fee, _paymentToken: feeToken});
```
[the remark this answers is not in the document]
Comment (JaredBorders) [reply src/Account.sol:774]: yes, when this check (see below) returns true in v2.0.1, the function returns and Gelato is never paid a fee (resulting in a failed tx on their end)

 (https://github.com/Kwenta/margin-manager/blob/22aecf8174555f93e17031458264b2815dd3ba58/src/Account.sol#L789)

File: src/Account.sol (hunk 1 above)
Comment (JaredBorders) [reply src/Account.sol:964]: Ah yes like before. Will add

**Review by JChiaramonte7 — COMMENTED** [review]
can't approve my own PR, but LGTM ✅

**Review by tommyrharper — APPROVED** [review]
LGTM 👍

**Pull request merged**